File: eco-core/core-plugin/src/main/java/com/willfp/eco/spigot/integrations/mcmmo/McmmoIntegrationImpl.java
package com.willfp.eco.spigot.integrations.mcmmo;

import com.gmail.nossr50.datatypes.meta.BonusDropMeta;
import com.gmail.nossr50.events.fake.FakeEvent;
import com.gmail.nossr50.mcMMO;
import com.willfp.eco.util.ClassUtils;
import com.willfp.eco.core.integrations.mcmmo.McmmoWrapper;
import org.bukkit.block.Block;
import org.bukkit.event.Event;
import org.jetbrains.annotations.NotNull;

public class McmmoIntegrationImpl implements McmmoWrapper {
    /**
     * Disabled if mcmmo is outdated or missing classes.
     */
    boolean disabled = false;

    /**
     * Initialize mcMMO integration.
     */
    public McmmoIntegrationImpl() {
        if (!ClassUtils.exists("com.gmail.nossr50.events.fake.FakeEvent")) {
            disabled = true;
        }
    }

    @Override
    public int getBonusDropCount(@NotNull final Block block) {
        if (disabled) {
            return 0;
        }

        if (block.getMetadata(mcMMO.BONUS_DROPS_METAKEY).size() > 0) {
            BonusDropMeta bonusDropMeta = (BonusDropMeta) block.getMetadata(mcMMO.BONUS_DROPS_METAKEY).get(0);
            return bonusDropMeta.asInt();
        }

        return 0;
    }

    @Override
    public boolean isFake(@NotNull final Event event) {
        if (disabled) {
            return false;
        }

        return event instanceof FakeEvent;
    }
}

File: eco-core/core-plugin/src/main/kotlin/com/willfp/eco/internal/spigot/integrations/mcmmo/McmmoIntegrationImpl.kt
package com.willfp.eco.internal.spigot.integrations.mcmmo

import com.gmail.nossr50.datatypes.meta.BonusDropMeta
import com.gmail.nossr50.events.fake.FakeEvent
import com.gmail.nossr50.util.MetadataConstants
import com.willfp.eco.core.integrations.mcmmo.McmmoIntegration
import com.willfp.eco.util.ClassUtils
import org.bukkit.block.Block
import org.bukkit.event.Event

class McmmoIntegrationImpl : McmmoIntegration {
    private var disabled = false
    override fun getBonusDropCount(block: Block): Int {
        if (disabled) {
            return 0
        }
        if (block.getMetadata(MetadataConstants.METADATA_KEY_BONUS_DROPS).isNotEmpty()) {
            val bonusDropMeta = block.getMetadata(MetadataConstants.METADATA_KEY_BONUS_DROPS)[0] as BonusDropMeta
            return bonusDropMeta.asInt()
        }
        return 0
    }

    override fun isFake(event: Event): Boolean {
        return if (disabled) {
            false
        } else event is FakeEvent
    }

    override fun getPluginName(): String {
        return "mcMMO"
    }

    init {
        if (!ClassUtils.exists("com.gmail.nossr50.events.fake.FakeEvent")) {
            disabled = true
        }
    }
}